### PR TITLE
Np2589 get all customers jsonld response

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDto.java [hunk 1]
```diff
@@ -175,34 +39,26 @@ public boolean equals(Object o) {
...
-            && Objects.equals(getDisplayName(), that.getDisplayName())
-            && Objects.equals(getShortName(), that.getShortName())
-            && Objects.equals(getArchiveName(), that.getArchiveName())
-            && Objects.equals(getCname(), that.getCname())
-            && Objects.equals(getInstitutionDns(), that.getInstitutionDns())
-            && Objects.equals(getFeideOrganizationId(), that.getFeideOrganizationId())
-            && Objects.equals(getCristinId(), that.getCristinId())
-            && Objects.equals(getVocabularySettings(), that.getVocabularySettings())
-            && Objects.equals(getContext(), that.getContext());
+                && Objects.equals(getIdentifier(), that.getIdentifier())
```
Comment (brinxmat) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDto.java:42]: Er dette riktig? Tipper at regen gjør noe her…kan ta feil

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDto.java (hunk 1 above)
Comment (janbj) [reply customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDto.java:42]: Hva tenker du på?

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerList.java [hunk 2]
```diff
@@ -18,34 +22,25 @@ public CustomerList() {
 
     }
 
-    public CustomerList(List<CustomerDto> customers) {
+    public CustomerList(URI id, List<CustomerDtoWithoutContext> customers) {
```
Comment (axthosarouris) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerList.java:25]: I think it would be simpler if CustomerList accepted CustomerDtos.
```
    public CustomerList(List<CustomerDto> customers) {
        this.id = NAMESPACE;
        this.customers = extractCustomers(customers);
        this.context = CustomerMapper.CONTEXT;
    }
    private List<CustomerDtoWithoutContext> extractCustomers(List<CustomerDto> customers) {
        return Optional.ofNullable(customers)
            .stream()
            .filter(Objects::nonNull)
            .flatMap(Collection::stream)
            .filter(Objects::nonNull)
            .map(CustomerDto::withoutContext)
            .collect(Collectors.toList());
    }
```

Comment (axthosarouris) [reply customer-commons/src/main/java/no/unit/nva/customer/model/CustomerList.java:25]: There is a problem with a test expecting a non empty list containing a null object.

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerMapper.java [hunk 3]
```diff
@@ -17,6 +24,22 @@
     private CustomerMapper() {
     }
 
+    private static CustomerDtoWithoutContext toCustomerDtoWithoutContext(CustomerDto customerDto) {
```
Comment (axthosarouris) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerMapper.java:27]: this belongs to object CustomerDto. 
```
 
    public CustomerDtoWithoutContext withoutContext() {
        return attempt(() -> JsonUtils.objectMapper.convertValue(this, CustomerDtoWithoutContext.class)).orElseThrow();
    }
```

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerMapper.java [hunk 4]
```diff
@@ -17,6 +24,22 @@
...
+        return objectMapper.convertValue(customerDto, CustomerDtoWithoutContext.class);
+    }
+
+    public static CustomerList toCustomerListFromCustomerDtosWithoutContexts(
+            List<CustomerDtoWithoutContext> customerDtos) {
+        URI id = URI.create(getIdNamespace());
+        return new CustomerList(id, customerDtos);
```
Comment (axthosarouris) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerMapper.java:34]: id should not be passed as a parameter. NAMESPACE is a public static variable of the system.

File: customer-commons/src/test/java/no/unit/nva/customer/model/CustomerTest.java [hunk 5]
```diff
@@ -47,6 +49,22 @@ public void customerMapperCanMapBetweenCustomerDtoAndCustomerDb() {
...
+        assertNotNull(customerList);
+    }
+
+    @Test
+    public void customerMapperCanMapCustomerDbToCustomerDto() {
+        CustomerDb customerDb = createCustomerDb();
+        CustomerDto customerDto = customerDb.toCustomerDto();
+        assertNotNull(customerDto);
+        assertNotNull(customerDto.getContext());
+    }
```
Comment (axthosarouris) [inline customer-commons/src/test/java/no/unit/nva/customer/model/CustomerTest.java:66]: this test is covered in CustomerDbTest. Just add a the test that the context is not null

**Review by axthosarouris — CHANGES_REQUESTED** [review]

**Comment by axthosarouris** [comment]
I do not understand this test:
```
    @Test
    public void customerListFromNull() {
        List<CustomerDto> list = new ArrayList<>();
        list.add(null);
        CustomerList customerList = new CustomerList(list);
        assertEquals(1, customerList.getCustomers().size());
        assertTrue(customerList.getCustomers().get(0) == null);
    }
```
Why do we want a non empty list with a null object instead of an empyt list.

**Review by brinxmat — DISMISSED** [review]

**Review by axthosarouris — APPROVED** [review]
